app/assets/javascripts/darkswarm/services/stripe_elements.js.coffee

```diff
@@ -15,6 +15,7 @@ Darkswarm.factory 'StripeElements', ($rootScope, Loading, RailsFlashLoader) ->
         if(response.error)
           Loading.clear()
           RailsFlashLoader.loadFlash({error: t("error") + ": #{response.error.message}"})
+          $rootScope.$apply()
```

Comment:
Was this added because it wasn't working in the browser, or because it wasn't working in the javascript spec?

Reply:
in the browser! this is the main thing in this PR, it fixes #5075

Reply:
Calling `$apply()` manually generally makes Angular a lot more brittle. Angular wraps almost everything in an `$apply()` call by default, and if `$apply()` ever gets called inside another `$apply()` call for any reason, it throws a fatal error. So I'm a bit nervous about putting it in critical checkout methods...

If it's absolutely needed, there are some safer options that will achieve the same thing, but won't ever throw fatal errors: wrap it in a `$timeout()` call (with zero "time"), or I think `$scope.$evalAsync()` does something similar but possibly in a more efficient way...

More info here: https://www.bennadel.com/blog/2605-scope-evalasync-vs-timeout-in-angularjs.htm

Reply:
I wish we could get rid of Angular... :see_no_evil:

Reply:
nice one Matt.
both apply with timeout and evalAsync fix the problem as well.
I thought evalAsync is nice (if we believe what the article explains) but it is not that known to other devs as $apply(), so I decided to use it but extract a method to explain why it is there.
Thanks!

Reply:
here
https://github.com/openfoodfoundation/openfoodnetwork/pull/5076/commits/957b398a549a569d9946ba6fff720dfc057d7664

Reply:
Ah, and it works? I thought the code to be run had to be passed in as a callback? Like: 
```
$rootScope.$evalAsync( ->
   Loading.clear()
   RailsFlashLoader.loadFlash({error: t("error") + ": #{response.error.message}"})
)
```
:man_shrugging:

Reply:
As long as we're not directly calling `$apply` here, I'm happy :+1: :+1: